sql/core/src/main/java/org/apache/spark/sql/execution/datasources/orc/OrcArrayColumnVector.java

```diff
@@ -52,61 +53,61 @@ public ColumnarArray getArray(int rowId) {
 
   @Override
   public boolean getBoolean(int rowId) {
-    throw new UnsupportedOperationException();
+    throw SparkUnsupportedOperationException.apply();
```

Comment:
There might be some omissions, such as:

https://github.com/apache/spark/blob/bc889c8c2adb976bdf3e7f020e3c7c3de5339c54/sql/core/src/main/java/org/apache/spark/sql/execution/datasources/orc/OrcAtomicColumnVector.java#L74

And there are some uses of the `UnsupportedOperationException` constructor with non-null arguments in the current code, such as: 

https://github.com/apache/spark/blob/bc889c8c2adb976bdf3e7f020e3c7c3de5339c54/sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedRleValuesReader.java#L787-L795

https://github.com/apache/spark/blob/bc889c8c2adb976bdf3e7f020e3c7c3de5339c54/sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/ColumnVectorUtils.java#L185

Are they not within the scope of modifications for the current PR?

Reply:
> Are they not within the scope of modifications for the current PR?

They are in the scope of this PR. I addressed your comments and replaces others exceptions in *.java. Thanks.